### PR TITLE
[#42] Workaround for wrong content type returned from jCenter

### PR DESCRIPTION
**Pull request by szpak**
Parser for 'application/unknown' is added to treat it as XML.

### CODE REVIEW

File: src/test/groovy/com/ofg/uptodate/finder/maven/MavenNewVersionFinderSpec.groovy [hunk 1]
```diff
@@ -59,9 +60,11 @@ class MavenNewVersionFinderSpec extends NewFinderSpec {
             1 * loggerProxy.lifecycle(_, NO_NEW_VERSIONS_MESSAGE)
     }
 
+    @Unroll
     def "should not fail or display excluded versions for dependencies found in external repo"() {
```
Comment (kamilszymanski) [inline src/test/groovy/com/ofg/uptodate/finder/maven/MavenNewVersionFinderSpec.groovy]: unroll name pattern (`#artifactVersion`) is missing

File: src/main/groovy/com/ofg/uptodate/finder/jcenter/JCenterNewVersionFinderFactory.groovy [hunk 2]
```diff
@@ -50,6 +51,11 @@ class JCenterNewVersionFinderFactory implements NewVersionFinderFactory {
         } as Future
     }
 
+    //application/unknown is returned from jCenter when using CloudFront - #42
+    private void handleUnknownContentTypeAsXml(HTTPBuilder httpBuilder) {
```
Comment (kamilszymanski) [inline src/main/groovy/com/ofg/uptodate/finder/jcenter/JCenterNewVersionFinderFactory.groovy]: Content type is not `unknown` or not known but `application/unknown`, let's reflect that in the method name

File: src/test/groovy/com/ofg/uptodate/finder/jcenter/JCenterNewVersionFinderFactoryWrongContentTypeSpec.groovy [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        project.extensions.uptodate.jCenterRepo = "http://localhost:${MOCK_HTTP_SERVER_PORT}/"
+        project.extensions.uptodate.ignoreMavenCentral = true
+    }
+
+    @Override
+    protected void artifactMetadataRequestResponse(String group, String name, String response) {
+        stubInteractionForJcenter(group, name, response, "application/unknown")
+    }
+
+    def "should list all dependencies, that have newer versions"() {
```
Comment (kamilszymanski) [inline src/test/groovy/com/ofg/uptodate/finder/jcenter/JCenterNewVersionFinderFactoryWrongContentTypeSpec.groovy]: how about including some information about content type in method name (or in `artifactMetadataRequestResponse` method name)?

Comment (szpak) [inline src/test/groovy/com/ofg/uptodate/finder/jcenter/JCenterNewVersionFinderFactoryWrongContentTypeSpec.groovy]: It is test copied from a base scenario. I forgot to rename it.

File: src/test/groovy/com/ofg/uptodate/finder/maven/MavenNewVersionFinderSpec.groovy (hunk 1 above)
Comment (szpak) [inline src/test/groovy/com/ofg/uptodate/finder/maven/MavenNewVersionFinderSpec.groovy]: Good catch. Nevertheless I don't have a good name for a text to display. Maybe:
```
@Unroll("#artifactVersion new version should be ignored when excluded")
def "should not fail or display excluded versions for dependencies found in external repo"() {
```

**Comment by szpak** [comment]
Rebased after code review.

File: src/test/groovy/com/ofg/uptodate/finder/jcenter/JCenterNewVersionFinderFactoryWrongContentTypeSpec.groovy [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        project.extensions.uptodate.jCenterRepo = "http://localhost:${MOCK_HTTP_SERVER_PORT}/"
+        project.extensions.uptodate.ignoreMavenCentral = true
+    }
+
+    @Override
+    protected void artifactMetadataRequestResponse(String group, String name, String response) {
+        stubInteractionForJcenter(group, name, response, "application/unknown")
+    }
+
+    def "should list all dependencies even when jcenter returned application/unknown content type"() {
```
Comment (kamilszymanski) [inline src/test/groovy/com/ofg/uptodate/finder/jcenter/JCenterNewVersionFinderFactoryWrongContentTypeSpec.groovy]: `should list new dependencies / dependency updates` would be more appropriate than `should list all dependencies`

**Comment by kamilszymanski** [comment]
+1

**Pull request merged**